docs/guides/production.md

```diff
@@ -0,0 +1,63 @@
+<!--[meta]
+section: guides
+title: Production Readiness Checklist
+[meta]-->
+
+# Production Readiness Checklist
+
+Yes, Keystone can be (and is!) used for production websites. Here's a handy list of tips for using Keystone with real workloads:
+
+## Session Handling
+
+### Cookie Secret
+
+Make sure the production deployment sets a long, unguessable value for [Keystone's `cookieSecret`](https://v5.keystonejs.com/keystone-alpha/keystone/#config).
+
+A randomly generated value is suitable (but keep it secret):
+
+```
+openssl rand -hex 32
+```
+
+### Session Store
+
+Sessions are stored inside the Keystone app by default, but in production it's recommended to store them in an external server such as Redis instead. You can use [any of the stores that work with `express session`](https://github.com/expressjs/session#compatible-session-stores). The advantages of using an external server are that
+
+- You can restart your app for upgrades without breaking sessions
+- You can replicate your Keystone app for availability, while keeping sessions consistent
+
+## Caching
+
+Improve performance and responsiveness by adding cache hints to your [lists](https://v5.keystonejs.com/api/create-list/#cachehint) and [fields](https://v5.keystonejs.com/keystone-alpha/fields/#cachehint).
+
+## Access Control
+
+Configure [access control](https://v5.keystonejs.com/guides/access-control/) to limit who can do what with your data.
+
+## DoS Hardening
+
+Add [query limits](https://v5.keystonejs.com/api/create-list/#querylimits) and [validation](https://v5.keystonejs.com/api/validation/) to protect your server against maliciously complex queries.
+
+## Using Reverse Proxies
+
+It's recommended to run production Javascript servers behind a reverse proxy such as [Nginx](https://nginx.org/), [HAProxy](https://www.haproxy.org/), a CDN or a cloud-based application (layer 7) load balancer. Doing that can improve performance and protect against [Slowloris Dos attacks](https://en.wikipedia.org/wiki/Slowloris_(computer_security)).
+
+You'll need to let your Javascript web app know that you're running behind a proxy, so that it can correctly handle `X-Forwarded-For-*` HTTP headers. Here's how you might set up an Express app to work behind one proxy:
+
+```javascript
+const app = express();
+app.set('trust proxy', 1); // behind one reverse proxy
+app.use(keystoneMiddlewares);
+```
```

Comment:
Can we explain a little more about how to configure this? Is it a custom server? If we're suggesting everyone needs this maybe we make it easy to set this middleware option on the Keystone constructor?

Reply:
Maybe you know more about this than me :)  Vocal runs KS on top of Express.  That `app.set()` line is the only thing we need to make Vocal work behind proxies, so everything else is just "normal" KS setup, whatever that's meant to be.

Reply:
A 'normal' set-up has no custom app like this :/

Reply:
Looks like a place where #1683 could come in handy

Reply:
Thanks, I just pushed an update, but I'll take a look at your PR to see if it's a better option.

Reply:
The options in #1683 are added to middlewares not the default express server

Reply:
Oh, sorry. Didn't realize there was a difference...  D: Should I make it set the default one too?

Reply:
That's a slightly trickier pattern since the default server is not part of the keystone instance and that's what @sarneaud solved for us with `configureServerApp(app)` where we explicitly pass in the app before applying the options.

We've done this specifically for `trust proxies` because we think it's a narrow and specific case that users may encounter without wanting a custom server.

The correct API for general configuration of the main server's express app remains a custom server. I don't want to introduce an alternative API for this.

I have similar reservations about #1683 and although I can appreciate your perspective but I think the correct API for significant modifications of an app is to create a custom App. I don't really want to open up the options the way we do in #1683. I think if possible having one way to do things is preferable.